opensrp-pnc/src/main/java/org/smartregister/pnc/repository/PncPartialFormRepository.java

```diff
@@ -0,0 +1,96 @@
+package org.smartregister.pnc.repository;
+
+import android.content.ContentValues;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import net.sqlcipher.Cursor;
+import net.sqlcipher.database.SQLiteDatabase;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.smartregister.pnc.dao.PncGenericDao;
+import org.smartregister.pnc.pojo.PncPartialForm;
+import org.smartregister.pnc.utils.PncDbConstants;
+import org.smartregister.repository.BaseRepository;
+
+import java.util.List;
+
+public class PncPartialFormRepository extends BaseRepository implements PncGenericDao<PncPartialForm> {
+
+    private static final String CREATE_TABLE_SQL = "CREATE TABLE " + PncDbConstants.Table.PNC_PARTIAL_FORM + "("
+            + PncDbConstants.Column.PncPartialForm.ID + " INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,"
+            + PncDbConstants.Column.PncPartialForm.BASE_ENTITY_ID + " VARCHAR NOT NULL, "
+            + PncDbConstants.Column.PncPartialForm.FORM + " TEXT NOT NULL, "
+            + PncDbConstants.Column.PncPartialForm.CREATED_AT + " INTEGER NOT NULL ," +
+            "UNIQUE(" + PncDbConstants.Column.PncPartialForm.BASE_ENTITY_ID + ") ON CONFLICT REPLACE)";
+
+    private static final String INDEX_BASE_ENTITY_ID = "CREATE INDEX " + PncDbConstants.Table.PNC_PARTIAL_FORM
+            + "_" + PncDbConstants.Column.PncPartialForm.BASE_ENTITY_ID + "_index ON " + PncDbConstants.Table.PNC_PARTIAL_FORM +
+            "(" + PncDbConstants.Column.PncPartialForm.BASE_ENTITY_ID + " COLLATE NOCASE);";
+
+    private String[] columns = new String[]{
+            PncDbConstants.Column.PncPartialForm.ID,
+            PncDbConstants.Column.PncPartialForm.BASE_ENTITY_ID,
+            PncDbConstants.Column.PncPartialForm.FORM,
```

Comment:
i think there is a column called form type

Reply:
added